File: MoonSimulation/Renderers/SunRenderer.cs
using SkiaSharp;

namespace MoonSimulation.Renderers;

/// <summary>
/// Renders the Sun with corona glow and sun rays.
/// </summary>
public static class SunRenderer
{
    /// <summary>
    /// Draws the Sun with corona glow and a happy face.
    /// </summary>
    public static void DrawSun(SKCanvas canvas, SKPoint center, float radius)
    {
        // Outer glow layers
        for (int i = 3; i >= 0; i--)
        {
            float glowRadius = radius * (1.5f + i * 0.4f);
            byte alpha = (byte)(30 - i * 6);
            using var glowPaint = new SKPaint
            {
                IsAntialias = true,
                Color = new SKColor(255, 200, 50, alpha),
                MaskFilter = SKMaskFilter.CreateBlur(SKBlurStyle.Normal, glowRadius * 0.3f)
            };
            canvas.DrawCircle(center, glowRadius, glowPaint);
        }

        // Sun body with radial gradient
        using var sunPaint = new SKPaint { IsAntialias = true };
        using var sunShader = SKShader.CreateRadialGradient(
            new SKPoint(center.X - radius * 0.2f, center.Y - radius * 0.2f),
            radius * 1.5f,
            new SKColor[] {
                new(255, 255, 220),
                new(255, 220, 80),
                new(255, 160, 20),
                new(200, 100, 0)
            },
            new float[] { 0f, 0.3f, 0.7f, 1f },
            SKShaderTileMode.Clamp);

        sunPaint.Shader = sunShader;
        canvas.DrawCircle(center, radius, sunPaint);

        // Happy face on Sun
        FaceRenderer.DrawCuteFace(canvas, center, radius, "happy");
    }

    /// <summary>
    /// Draws subtle sun ray lines emanating from the sun.
    /// </summary>
    public static void DrawSunRays(SKCanvas canvas, SKPoint center, float radius)
    {
        using var rayPaint = new SKPaint
        {
            IsAntialias = true,
            Style = SKPaintStyle.Stroke,
            StrokeWidth = 1.5f
        };

        for (int i = 0; i < 12; i++)
        {
            float angle = i * MathF.PI * 2 / 12;
            float innerR = radius * 1.3f;
            float outerR = radius * 1.7f;

            float x1 = center.X + MathF.Cos(angle) * innerR;
            float y1 = center.Y + MathF.Sin(angle) * innerR;
            float x2 = center.X + MathF.Cos(angle) * outerR;
            float y2 = center.Y + MathF.Sin(angle) * outerR;

            byte alpha = (byte)(40 + 20 * MathF.Sin(angle * 3));
            rayPaint.Color = new SKColor(255, 230, 100, alpha);
            canvas.DrawLine(x1, y1, x2, y2, rayPaint);
        }
    }
}

File: web/js/renderers/sunRenderer.js
import { drawCuteFace } from './faceRenderer.js';

// Renders the Sun with corona glow, rays, and a happy face.

export function drawSun(ctx, cx, cy, radius) {
  // Outer glow layers
  for (let i = 3; i >= 0; i--) {
    const glowRadius = radius * (1.5 + i * 0.4);
    const alpha = (30 - i * 6) / 255;
    ctx.save();
    ctx.shadowBlur = glowRadius * 0.3;
    ctx.shadowColor = `rgba(255,200,50,${alpha})`;
    ctx.fillStyle = `rgba(255,200,50,${alpha})`;
    ctx.beginPath();
    ctx.arc(cx, cy, glowRadius, 0, Math.PI * 2);
    ctx.fill();
    ctx.restore();
  }

  // Sun body with radial gradient
  const grad = ctx.createRadialGradient(
    cx - radius * 0.2, cy - radius * 0.2, 0,
    cx - radius * 0.2, cy - radius * 0.2, radius * 1.5
  );
  grad.addColorStop(0, 'rgb(255,255,220)');
  grad.addColorStop(0.3, 'rgb(255,220,80)');
  grad.addColorStop(0.7, 'rgb(255,160,20)');
  grad.addColorStop(1, 'rgb(200,100,0)');

  ctx.fillStyle = grad;
  ctx.beginPath();
  ctx.arc(cx, cy, radius, 0, Math.PI * 2);
  ctx.fill();

  // Happy face on Sun
  drawCuteFace(ctx, cx, cy, radius, 'happy');
}

export function drawSunRays(ctx, cx, cy, radius) {
  ctx.save();
  ctx.lineWidth = 1.5;
  ctx.lineCap = 'round';

  for (let i = 0; i < 12; i++) {
    const angle = (i * Math.PI * 2) / 12;
    const innerR = radius * 1.3;
    const outerR = radius * 1.7;

    const x1 = cx + Math.cos(angle) * innerR;
    const y1 = cy + Math.sin(angle) * innerR;
    const x2 = cx + Math.cos(angle) * outerR;
    const y2 = cy + Math.sin(angle) * outerR;

    const alpha = (40 + 20 * Math.sin(angle * 3)) / 255;
    ctx.strokeStyle = `rgba(255,230,100,${alpha})`;
    ctx.beginPath();
    ctx.moveTo(x1, y1);
    ctx.lineTo(x2, y2);
    ctx.stroke();
  }
  ctx.restore();
}
